halo/evmstaking2/keeper/keeper.go

```diff
@@ -240,6 +240,10 @@ func (k Keeper) deliverDelegate(ctx context.Context, ev *bindings.StakingDelegat
 		return errors.New("validator does not exist", "validator", valAddr.String())
 	}
 
+	if ev.Amount == nil {
```

Comment:
lets add a "verifyStakingDelegate` function. Also, we can rename the `ev` variable, it was copy paste bug.